app/services/teacher_training_public_api/sync_all_providers_and_courses.rb

```diff
@@ -18,12 +22,12 @@ def self.call(recruitment_cycle_year: ::RecruitmentCycle.current_year)
       raise TeacherTrainingPublicAPI::SyncError
     end
 
-    def self.sync_providers(providers_from_api, recruitment_cycle_year)
+    def self.sync_providers(providers_from_api, recruitment_cycle_year, incremental_sync: true)
```

Comment:
Why do we need to pass this down here — won't it be the case that only providers that have been touched will be included in `providers_from_api`?

Reply:
The flag gets passed down to `TeacherTrainingPublicAPI::SyncCourses` - I figured given a Provider can be updated independently of its courses we may as well optimise course sync also?

Reply:
I think 90% of the benefit will come from reducing the number of providers whose courses we have to look at. But if we've done the work, great!